contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+            }
+            _leveragedTokenData.collateralCToken = components[0];
+            requiredCTokenAmount = equityPositions[0];
+            _leveragedTokenData.debtToken = components[1];
+            _leveragedTokenData.debtAmount = debtPositions[1];
+        } else {
+            if (components[1] == cEtherADDRESS) {
+                _collateralToken = addresses.weth;
+            } else {
+                _collateralToken = CErc20Storage(components[1]).underlying();
+            }
+            _leveragedTokenData.collateralCToken = components[1];
+            requiredCTokenAmount = equityPositions[1];
+            _leveragedTokenData.debtToken = components[0];
+            _leveragedTokenData.debtAmount = debtPositions[0];
+        }
+        Exp memory exchangeRate = Exp({mantissa: ICEther(payable(_leveragedTokenData.collateralCToken)).exchangeRateStored()});
+        uint256 collateralAmount = mul_(requiredCTokenAmount, exchangeRate);
+        _leveragedTokenData.collateralAmount = collateralAmount + ROUNDING_ERROR_MARGIN;
+        _leveragedTokenData.collateralToken = _collateralToken;
+        return _leveragedTokenData; 
+    }
+
+
+
+    /**
+     * Approves max amount of given token to all exchange routers and the debt issuance module
+     *
+     * @param _token  Address of the token to be approved
+     */
+    function _approveToken(IERC20 _token) internal {
+        _safeApprove(_token, address(debtIssuanceModule), MAX_UINT256);
+    }
+
+    /**
+     * Initiates a flashloan call with the correct parameters for issuing set tokens in the callback
+     * Borrows correct amount of collateral token and and forwards encoded memory to controll issuance in the callback.
+     *
+     * @param _setToken                     Address of the SetToken being initialized
+     * @param _setAmount                    Amount of the SetToken being initialized
+     * @param _inputToken                   Address of the input token to pay with
+     * @param _maxAmountInputToken          Maximum amount of input token to pay
+     * @param _swapDataDebtForCollateral    Data (token addresses and fee levels) to describe the swap path from Debt to collateral token
+     * @param _swapDataInputToken           Data (token addresses and fee levels) to describe the swap path from input to collateral token
+     */
+    function _initiateIssuance(
+        ISetToken _setToken,
+        uint256 _setAmount,
+        address _inputToken,
+        uint256 _maxAmountInputToken,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken
+    )
+    internal
+    {
+        // need to check (true or false to issue)
+        compoundLeverageModule.sync(_setToken, true);
+
+        LeveragedTokenData memory leveragedTokenData = _getLeveragedTokenData(_setToken, _setAmount, true);
+
+        // get cTokenAddress from underlying assets
+        _getCTokenAddess(leveragedTokenData.collateralToken);
```

Comment:
You don't use the returned value here so I guess you can remove this statement.

Reply:
addressed in 2bc43dfd123d9c826c22e1931048af595bb3ebc3